src/Microsoft.ApplicationInsights/Metrics/MetricIdentifier.cs

```diff
@@ -33,23 +35,33 @@ public static string DefaultMetricNamespace
 
             set
             {
-                ValidateLiteral(value, nameof(value));
+                ValidateLiteral(value, nameof(value), allowEmpty: true);
```

Comment:
I haven't noticed this class before. Basically it is a global singleton that allows to override the default namespace for the process?

It may not work well for aspnetcore apps which uses DI for per-app configuration. And multiple apps may be configured in a single process. Perhaps this setting can be moved to `telemetryconfiguration`?

Reply:
Good point. Let's have a brief chat so that I can better understand why statics on Telemetry Config are better than here, and then I can get this moved accordingly.